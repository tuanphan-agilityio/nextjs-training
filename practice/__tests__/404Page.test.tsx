import React from 'react';
import { render, screen } from '@testing-library/react';
import NotFound from '@/pages/404';

describe('NotFound Page', () => {
  it('404 Page matches DOM Snapshot', () => {
    const { container } = render(<NotFound />);

    expect(container).toMatchSnapshot();
  });

  it('should have a link to the home page', () => {
    render(<NotFound />);

    const linkElement = screen.getByText('Go to Home Page');
    const href = linkElement.getAttribute('href');
    expect(href).toBe('/');
  });
});
