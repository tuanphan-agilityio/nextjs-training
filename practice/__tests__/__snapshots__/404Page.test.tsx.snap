// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page 404 Page matches DOM Snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-center h-screen"
  >
    <div
      class="pb-20 w-full text-center "
    >
      <h1
        class="text-3xl"
      >
        Oops!
      </h1>
      <h2
        class="text-secondary text-2xl"
      >
        404 - Page not found
      </h2>
      <p
        class="mb-5 text-secondary text-sm"
      >
        The page you are looking for might have been removed had its name changed or is temporarily unavailable.
      </p>
      <button
        class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary"
      >
        <a
          href="/"
        >
          Go to Home Page
        </a>
      </button>
    </div>
  </div>
</div>
`;
